### DISCUSSION TITLE
Automatic cleaning of `Finished uploads` log

### DISCUSSION THREAD

**Discussion by Korb** [airdcpp/airdcpp-windows#161] [Q&A]
It seems to me that I have automatic cleaning of `Finished uploads` log enabled: the `Finished uploads` tab is empty even if I am uploading small files, i.e. actually completed uploads should be present in the log.

I can't find this setting. Where is it hiding?

AirDC++ 4.21
Microsoft Windows 10.0.19045.5073

**Comment by Sopor** [comment]
You can only see `Finished uploads` per session.

When you close AirDC it will be removed from the `Finished uploads` window.

If you want to keep a log of every uploads you can enable it in Logs, but it will still not show all uploads in the `Finished uploads` window.

You have to open the Uploads.log in a text editor or viewer.

![image](https://github.com/user-attachments/assets/c9881b12-52d3-4d6c-854c-b069cd8ea0d1)

**Comment by Korb** [comment]
@Sopor, thanks! Almost 100,000 lines in three weeks - that would be hard to display without slowing down the interface.

So, there is no setting in the application that would prevent the current log from being displayed in the `Finished uploads` tab, but would allow events to be written to the `Uploads.log` file? For example, unchecking the `Show finished uploads` checkbox in the `Settings` / `Advanced` / `Logs` / `System log / statusbar` block.

**Comment by Sopor** [comment]
I'm not really sure i understand what you mean...
If you don't want to see `Finished uploads`, don't open it 😊

If you uncheck `Show finished uploads` it will not be written to the `system.log` file.
